temporal/t.rast.what/t.rast.what.py

```diff
@@ -373,78 +373,72 @@ def one_point_per_row_output(
-                if vcat:
-                    cat_str = "{ca}{sep}".format(ca=cat, sep=separator)
-                else:
-                    cat_str = ""
-                if site_input:
-                    coor_string = (
-                        "%(x)10.10f%(sep)s%(y)10.10f%(sep)s%(site_name)s%(sep)s"
-                        % (
+                    else:
+                        x = line[0]
+                        y = line[1]
+                        if site_input:
+                            site = line[2]
+                        values = line[3:]
+
+                    for i in range(len(values)):
+                        start, end = map_list[i].get_temporal_extent_as_tuple()
+                        if vcat:
+                            cat_str = "{ca}{sep}".format(ca=cat, sep=separator)
+                        else:
+                            cat_str = ""
+                        if site_input:
+                            coor_string = (
+                                "%(x)10.10f%(sep)s%(y)10.10f%(sep)s%(site_name)s%(sep)s"
+                                % (
+                                    {
+                                        "x": float(x),
+                                        "y": float(y),
+                                        "site_name": str(site),
+                                        "sep": separator,
+                                    }
+                                )
+                            )
+                        else:
+                            coor_string = "%(x)10.10f%(sep)s%(y)10.10f%(sep)s" % (
+                                {"x": float(x), "y": float(y), "sep": separator}
+                            )
+                        time_string = "%(start)s%(sep)s%(end)s%(sep)s%(val)s\n" % (
                             {
-                                "x": float(x),
-                                "y": float(y),
-                                "site_name": str(site),
+                                "start": str(start),
+                                "end": str(end),
+                                "val": (values[i].strip()),
                                 "sep": separator,
                             }
                         )
-                    )
-                else:
-                    coor_string = "%(x)10.10f%(sep)s%(y)10.10f%(sep)s" % (
-                        {"x": float(x), "y": float(y), "sep": separator}
-                    )
-                time_string = "%(start)s%(sep)s%(end)s%(sep)s%(val)s\n" % (
-                    {
-                        "start": str(start),
-                        "end": str(end),
-                        "val": (values[i].strip()),
-                        "sep": separator,
-                    }
-                )
-
-                out_file.write(cat_str + coor_string + time_string)
 
-        in_file.close()
-
-    if out_file is not sys.stdout:
-        out_file.close()
```

Comment:
Is there a way to have a test that triggers this? When reading carefully my code coverage equivalent branch, the t.rast.what file was only imported. I think it's just a collection problem, as there is a test using stdout and doesn't specify layout, so row will be used, entering in this function.
I want to know what happens when having a sys.stdout used as with a context manager, does it close it when exiting the context? It didn't break any tests though.
To be sure of the syntax, in the with of out_file above, I used way more parentheses, and let black simplify them back. I also checked with small examples in the same function above that pyright detected usages that allow or not with a context manager, and sys.stdout can be used with a context manager. It works just like text files. Can it really be closed?

Reply:
There are three tests in `test_what.py` which test with `output="-"`, so that's likely testing this stdout case.

I didn't check, but I would guess that you _can_ close stdout, but you don't want to because any code later can be "surprised" by that.

Reply:
In another of your review comments, you showed what happened when closing sys.stdout, (not good). Testing with ipython, it makes ipython crash. So: to avoid

Reply:
Using `contextlib.nullcontext` to wrap sys.stdout, so should be ok now